experiments/tf_trainer/tf_hub_tfjs/notebook/BiasEvaluation.ipynb

```diff
@@ -0,0 +1,512 @@
+        "  ax.xaxis.tick_top()\n",
+        "  plt.xticks(rotation=90)\n",
+        "  ax.vlines(vlines, *ax.get_ylim())\n",
+        "  return ax\n",
+        "\n",
+        "plot_auc_heatmap(bias_metrics_df, [MODEL_NAME])"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "cyv6n0GxpCwb",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "# Plot histograms showing comment scores"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "AKm925FWX_Fd",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "We can graph a histogram of comment scores in each identity.  In the following graphs, the X axis represents the toxicity score given by our new model, and the Y axis represents the comment count.  Blue values are comment whose true label is non-toxic, while red values are those whose true label is toxic."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "iMSpM0U1YAN1",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def plot_histogram(non_toxic_scores, toxic_scores, description):\n",
+        "  NUM_BINS=10\n",
+        "  sns.distplot(non_toxic_scores, norm_hist=True, bins=NUM_BINS, color=\"skyblue\", label='non-toxic ' + description, kde=False)\n",
+        "  ax = sns.distplot(toxic_scores, norm_hist=True, bins=NUM_BINS, color=\"red\", label='toxic ' + description, kde=False)\n",
+        "  ax.set(xlabel='model toxicity score', ylabel='relative % of comments', yticklabels=[])\n",
+        "  plt.legend()\n",
+        "  plt.figure()\n",
+        "\n",
+        "# Plot toxicity distributions of different identities to visualize bias.\n",
+        "def plot_histogram_for_identity(df, identity):\n",
+        "  toxic_scores = df.query(identity + ' == True & toxicity == True')[MODEL_NAME]\n",
+        "  non_toxic_scores = df.query(identity + ' == True & toxicity == False')[MODEL_NAME]\n",
+        "  plot_histogram(non_toxic_scores, toxic_scores, 'labeled for ' + identity)\n",
+        "\n",
+        "def plot_background_histogram(df):\n",
+        "  toxic_scores = df.query('toxicity == True')[MODEL_NAME]\n",
+        "  non_toxic_scores = df.query('toxicity == False')[MODEL_NAME]\n",
+        "  plot_histogram(non_toxic_scores, toxic_scores, 'for all test data')\n",
+        "\n",
+        "# Plot the histogram for the background data, and for a few identities\n",
+        "plot_background_histogram(test_df)\n",
+        "plot_histogram_for_identity(test_df, 'heterosexual')\n",
+        "plot_histogram_for_identity(test_df, 'transgender')\n",
+        "plot_histogram_for_identity(test_df, 'homosexual_gay_or_lesbian')\n",
+        "plot_histogram_for_identity(test_df, 'atheist')\n",
+        "plot_histogram_for_identity(test_df, 'christian')\n",
+        "plot_histogram_for_identity(test_df, 'asian')"
```

Comment:
For the histograms, you might want to change them to your 3 with lowest scores (homosexual_gay_of_lesbian, transgender, black) and 3 with highest scores (christian, atheist, asian).  I think the examples you picked were copied from the FAT* model which had different results.

Reply:
Removed